handlers_test.go

```diff
@@ -660,6 +662,55 @@ func TestHandlers(t *testing.T) {
 	}
 }
 
+func TestCreateLogFilesHandler(t *testing.T) {
+	logWriterBuf := new(bytes.Buffer)
+	config := Config{
+		LogFifo:       filepath.Join(testDataPath, "firecracker-log.fifo"),
+		MetricsFifo:   filepath.Join(testDataPath, "firecracker-metrics.fifo"),
+		FifoLogWriter: logWriterBuf,
+	}
+
+	defer func() {
+		os.Remove(config.LogFifo)
+		os.Remove(config.MetricsFifo)
+	}()
+
+	ctx := context.Background()
+	m, err := NewMachine(ctx, config, WithLogger(fctesting.NewLogEntry(t)))
+	if err != nil {
+		t.Fatalf("failed to create machine: %v", err)
+	}
+
+	// spin off goroutine to write to Log fifo so we don't block
+	go func() {
+		timer := time.NewTimer(1 * time.Second)
+		for {
+			select {
+			case <-timer.C:
+				t.Error("timed out waiting for log fifo")
+			default:
+				fifoPipe, err := os.OpenFile(config.LogFifo, os.O_WRONLY, os.ModePerm)
+				if err != nil {
+					time.Sleep(10 * time.Millisecond)
+					continue
+				}
+
+				timer.Stop()
+				_, err = fifoPipe.WriteString("data")
+				if err != nil {
+					t.Errorf("Failed to write to fifo %v", err)
+				}
+				return
+			}
+		}
+	}()
+
+	err = CreateLogFilesHandler.Fn(ctx, m)
```

Comment:
I think it would be good to assert that you can read the string "data" from `config.LogFifo` after the handler was called.

You can use a channel to communicate that the write has occurred (inside the goroutine above), and then delay the assertion until that has happened.

Reply:
I added an assertion checking that we wrote to the fifo. I had to do some polling as it there was a bit of race condition between the writer copying from the fifo and us checking it's output.